pandas/io/parsers.py

```diff
@@ -1174,33 +1234,36 @@ def _make_engine(self, engine="c"):
             else:
                 raise ValueError(
                     f"Unknown engine: {engine} (valid options "
-                    'are "c", "python", or "python-fwf")'
+                    'are "c", "python", "pyarrow", or "python-fwf")'
                 )
             self._engine = klass(self.f, **self.options)
 
     def _failover_to_python(self):
         raise AbstractMethodError(self)
 
     def read(self, nrows=None):
-        nrows = _validate_integer("nrows", nrows)
-        ret = self._engine.read(nrows)
+        if self.engine == "pyarrow":
+            df = self._engine.read()
+        else:
+            nrows = _validate_integer("nrows", nrows)
```

Comment:
you could do this at the top of the function

Reply:
The problem here is that `nrows` is not supported by pyarrow, so in principle we should only need to validate it for other engines. 
Now, I suppose that before you end up here, we already would have raised an error if `nrows` was specified in case of the pyarrow engine, so it probably doesn't hurt to needlessly validate the default `nrows` argument in case of the pyarrow engine.

Reply:
Yeah, I guess I could do that, but I think my way is cleaner, since all the pyarrow code would be in the if, and the other parser code would be in the else.